src/set_relation/set_relation.cc

```diff
@@ -20,120 +20,74 @@
-
-  return imap;
-}
-
-/*! This function takes an isl_map* and returns pointer to equ. Relation string
-** The function takes ownership of input argument 'imap'
-*/
-std::string islMapToString ( isl_map* imap , isl_ctx *ctx )
-{
-  // Get an isl printer and associate to an isl context
-  isl_printer * ip = isl_printer_to_str(ctx);
-
-  // get string back from ISL map
-  isl_printer_set_output_format(ip , ISL_FORMAT_ISL);
-  isl_printer_print_map(ip ,imap);
-  char *i_str = isl_printer_get_str(ip);
-  std::string stringFromISL (i_str); 
-  
-  // clean-up
-  isl_printer_flush(ip);
-  isl_printer_free(ip);
-  isl_map_free(imap);
-  imap= NULL;
-  free(i_str);
-
-  return stringFromISL;
-}
-
-//! runs the Set through ISL and returns the resulting string
-string passSetThruISL(string rstr) {
+  string sstr = s->toISLString();
 
   isl_ctx *ctx = isl_ctx_alloc();
-
-  string result =  islSetToString ( islStringToSet(rstr,ctx), ctx );
-
+  string islStr =  islSetToString ( islStringToSet(sstr,ctx), ctx );
   isl_ctx_free(ctx);
 
+  // We need to revert changes that isl applies to Tuple Declaration
+  int inArity = s->arity(), outArity = 0;
+  string corrected = revertISLTupDeclToOrig( sstr, islStr, inArity, outArity);
+  Set* result = new Set(corrected);
+
   return result;
 }
 
-//! Runs a Relation string through ISL and returns the resulting string
-string passRelationThruISL(string rstr) {
+//! Runs an Affine Relation through ISL and returns the normalized result
+Relation* passRelationThruISL(Relation* r){
+
+  string rstr = r->toISLString();
 
   isl_ctx *ctx = isl_ctx_alloc();
+  string islStr =  islMapToString ( islStringToMap(rstr,ctx), ctx );
+  isl_ctx_free(ctx);
 
-  string result =  islMapToString ( islStringToMap(rstr,ctx), ctx );
+  // We need to revert changes that isl applies to Tuple Declaration
+  int inArity = r->inArity(), outArity = r->outArity();
+  string corrected = revertISLTupDeclToOrig( rstr, islStr, inArity, outArity);
+  Relation* result = new Relation( corrected);
 
-  isl_ctx_free(ctx);
+//    std::cout<<std::endl<<"r = "<<rstr<<std::endl<<"i = "<<islStr<<std::endl<<std::endl;
+//    std::cout<<std::endl<<"c = "<<corrected<<std::endl<<std::endl;
 
   return result;
 }
 
 // This function can be used for Projecting out a tuple variable
 // from an affine set string using isl library
-string islSetProjectOut(string rstr, unsigned pos) {
+Set* islSetProjectOut(Set* s, unsigned pos) {
 
-  isl_ctx *ctx = isl_ctx_alloc();
+  string sstr = s->toISLString();
 
-  string result = islSetToString ( 
-               isl_set_project_out(islStringToSet(rstr,ctx), 
+  isl_ctx *ctx = isl_ctx_alloc();
+  string islStr = islSetToString ( 
+               isl_set_project_out(islStringToSet(sstr,ctx), 
                                    isl_dim_out, pos, 1), ctx 
                );
-
   isl_ctx_free(ctx);
 
+  // We need to revert the changes that isl applies to Tuple Declaration
```

Comment:
Talk some about the more work is being done here that isn't done in passSetThruISL and passRelationThruISL.  Is there an opportunity to remove some code repetition?